MeshLib/NodePartitionedMesh.h

```diff
@@ -115,6 +120,19 @@ class NodePartitionedMesh : public Mesh
             return _n_nghost_elem;
         }
 
+        /// Get the maximum number of connected nodes to node.
+        std::size_t getMaximumNConnectedNodesToNode() const
+        {
+            std::vector<Node *>::const_iterator it_max_ncn = std::max_element(
+                _nodes.cbegin(), _nodes.cend(),
+                [](Node const *const node_a, Node const *const node_b)
+                {
+                    return (node_a->getConnectedNodes().size() <
+                            node_b->getConnectedNodes().size());
+                });
+            return (*it_max_ncn)->getConnectedNodes().size() + 1;
```

Comment:
The number of connected nodes should include the node itself. Besides, this number is used to preallocate memory for PETSc matrix, the number plus one guarantees an enough space for the sparsity of the matrix arising from FEM.